packages/server/graphql/mutations/helpers/bootstrapNewUser.ts

```diff
@@ -48,11 +79,13 @@ const bootstrapNewUser = async (
     experimentalFlags.push('retrosInDisguise')
   }
 
-  await Promise.all([
+  const usersWithDomainTeamIds = usersWithDomain.flatMap((user) => user.tms)
+  const [usersWithDomainTeams] = await Promise.all([
```

Comment:
-2 Starting from users with the same domain is the wrong approach. If I have an auto join team at bigcorp.com, I don't want any random users join just because I have joe@consultants.org on there as well.
We should start from the organization with active domain instead.

Reply:
Makes sense. I was checking whether the org the team belongs to has the same `activeDomain` as the user, so `consultants.org` wouldn't have joined `bigcorp.com`, but I agree that starting from the org makes sense. Initially, I thought that would be expensive, but I see it's an index, so it would be efficient.